src/editor/controller/editor_controller.cpp

```diff
@@ -876,6 +876,14 @@ bool editor_controller::do_execute_command(const hotkey::ui_command& cmd, bool p
 			un->anim_comp().set_standing();
 		}
 		return true;
+		case HOTKEY_EDITOR_UNIT_TOGGLE_LOYAL:
+		{
+			map_location loc = gui_->mouseover_hex();
+			const unit_map::unit_iterator un = get_current_map_context().units().find(loc);
+			bool loyal = un->loyal();
```

Comment:
What happens if no unit is found?

Reply:
The "Loyal" checkbox is not visible unless there is actually an unit on that hex (so this cannot be triggered). I'll try to add a failsafe.

Reply:
> The "Loyal" checkbox is not visible unless there is actually an unit on that hex (so this cannot be triggered).

You can bind the hotkey to something and trigger it with any hex selected.

Reply:
Still doesn't cause any problems (no crash or something).

Reply:
One thing I noticed though is that when that action is triggered by hotkey, the unit didn't redraw until I clicked somewhere else.

Reply:
If this is defined behavior then it's all good. If it's undefined behavior then not crashing means nothing.